ocaml/xapi/xapi_vm_migrate.ml

```diff
@@ -312,9 +312,33 @@ let pool_migrate ~__context ~vm ~host ~options =
   let queue_name = queue_of_vm ~__context ~self:vm in
   let module XenopsAPI = (val make_client queue_name : XENOPS) in
   let session_id = Ref.string_of (Context.get_session_id __context) in
-  let ip =
-    Http.Url.maybe_wrap_IPv6_literal (Db.Host.get_address ~__context ~self:host)
+  (* If `network` provided in `options`, try to get `xenops_url` on this network *)
+  let address =
+    match List.assoc_opt "network" options with
+    | None ->
+        Db.Host.get_address ~__context ~self:host
+    | Some network_ref -> (
+        let pifs =
+          Db.Network.get_PIFs ~__context ~self:(Ref.of_string network_ref)
+        in
+        try
+          let pif =
+            List.find (fun x -> host = Db.PIF.get_host ~__context ~self:x) pifs
+          in
+          match Db.PIF.get_primary_address_type ~__context ~self:pif with
+          | `IPv4 ->
+              Db.PIF.get_IP ~__context ~self:pif
+          | `IPv6 -> (
+            match Db.PIF.get_IPv6 ~__context ~self:pif with
+            | [] ->
+                Db.Host.get_address ~__context ~self:host
+            | ip :: _ ->
+                ip
+          )
+        with Not_found -> Db.Host.get_address ~__context ~self:host
```

Comment:
We should raise an error, because the client has explicitly asked to use a certain network.